x/evm/keeper/state_transition_vfc_bank.go

```diff
@@ -0,0 +1,297 @@
+	case types.VFBCmTransfer:
+		const opGasCost = types.VFBCopgTransfer
+		const opGasCostOnRevert = types.VFBCopgTransfer_Revert
+		if gas < opGasCost {
+			return types.NewExecVFCOutOfGas()
+		}
+
+		if len(calldata) != 4 /*4bytes sig*/ +32 /*address*/ +32 /*amount*/ {
+			return types.NewExecVFCRevert(opGasCostOnRevert, errors.New("invalid call data"))
+		}
+
+		eventTransfer, foundEvent := compiledVFContract.ABI.Events["Transfer"]
+		if !foundEvent {
+			return types.NewExecVFCError(errors.New("event Transfer could not be found"))
+		}
+
+		// unpack the call-data
+		inputs, err := compiledVFContract.UnpackInput("transfer", calldata[4:])
+
+		if err != nil {
+			return types.NewExecVFCError(err)
+		}
+
+		if len(inputs) != 2 {
+			return types.NewExecVFCRevert(opGasCostOnRevert, errors.New("invalid input"))
+		}
+
+		to, ok := inputs[0].(common.Address)
+		if !ok {
+			return types.NewExecVFCRevert(opGasCostOnRevert, errors.New("first input is not an address"))
+		}
+
+		amount, ok := inputs[1].(*big.Int)
+		if !ok {
+			return types.NewExecVFCRevert(opGasCostOnRevert, errors.New("second input is not a number"))
+		}
+
+		receiver := sdk.AccAddress(to.Bytes())
+
+		// prohibit transfer to some types of account
+
+		// - module account
+		accountI := k.accountKeeper.GetAccount(ctx, receiver)
+		if accountI != nil {
+			_, isModuleAccount := accountI.(authtypes.ModuleAccountI)
+			if isModuleAccount {
+				return types.NewExecVFCRevert(opGasCostOnRevert, errors.New("can not transfer to module account"))
+			}
+		}
+		// - VF contracts
+		if k.IsVirtualFrontierContract(ctx, to) {
+			return types.NewExecVFCRevert(opGasCostOnRevert, errors.New("not allowed to receive"))
+		}
+
+		senderBalance := k.bankKeeper.GetBalance(ctx, sender.Bytes(), bankContractMetadata.MinDenom)
+		sendAmount := sdk.NewCoin(bankContractMetadata.MinDenom, sdk.NewIntFromBigInt(amount))
+
+		if senderBalance.Amount.LT(sendAmount.Amount) {
+			return types.NewExecVFCRevert(opGasCostOnRevert, errors.New("ERC20: transfer amount exceeds balance"))
+		}
+
+		if err := k.bankKeeper.IsSendEnabledCoins(ctx, sendAmount); err != nil {
+			return types.NewExecVFCRevert(opGasCostOnRevert, errors.New(err.Error()))
+		}
+
+		if k.bankKeeper.BlockedAddr(receiver) {
+			return types.NewExecVFCRevert(opGasCostOnRevert, errors.New(fmt.Sprintf("unauthorized, %s is not allowed to receive funds", to)))
+		}
+
+		// Prepare to fire the ERC-20 Transfer event
+		bzData, err := abi.Arguments{
+			eventTransfer.Inputs[2],
+		}.Pack(amount)
+		if err != nil {
+			if err != nil {
+				return types.NewExecVFCError(err)
+			}
+		}
+
+		// The rest code are state changed, can not be reverted
+
+		// transfer the amount
+		if err := k.bankKeeper.SendCoins(ctx, sender.Bytes(), receiver, sdk.NewCoins(sendAmount)); err != nil {
+			return types.NewExecVFCRevert(opGasCost, errors.Wrap(err, "failed to transfer"))
+		}
+
+		// Fire the ERC-20 Transfer event
+		stateDB.AddLog(&ethtypes.Log{
```

Comment:
isn't it conflicting with the statedb.Commit() check 
```
if s.keeper.IsVirtualFrontierContract(s.ctx, addr) {
    return errorsmod.Wrapf(evmtypes.ErrProhibitedAccessingVirtualFrontierContract, ...
		}
```

Reply:
EVM is not invoked. State DB won't be invoked. VFC, we simulate the EVM, not invoking it.